internal/xds/bootstrap/bootstrap.go

```diff
@@ -275,6 +284,12 @@ func (sc *ServerConfig) MarshalJSON() ([]byte, error) {
 	return json.Marshal(server)
 }
 
+// dialer captures the Dialer method specified via the credentials bundle.
+type dialer interface {
+	// Dialer specifies how to dial the xDS server.
+	Dialer(context.Context, string) (net.Conn, error)
```

Comment:
Can this method name be more specific? Maybe something like `DialXDS` or `DialXDSServer` or something else?

The reason I'm suggesting this is because the API to register a credentials type is public. So, users *could* register their own credentials using `bootstrap.RegisterCredentials`, and if they end up having a `Dialer` method on their creds implementation type (which is maybe doing something else), then, they would be in for a surprise.

Or at least, we should document this in the public API in package `xds/bootstrap`?

Thoughts @dfawley

Reply:
Acknowledged.   
Users could definitely have a  the `Dialer` method on their creds bundle implementation and it is a valid concern. We have no idea or control of what users could implement. Using another "less common" name compared to `Dialer` would greatly reduce this risk.

I think `DialXDS` would be appropriate and not too verbose, although this would introduce slight repetition as the `XDS` suffix is already implied (per the Repetition section of the Google Go Style Decisions guide).

What are your thoughts @dfawley? If everyone is in agreement, then I will make the change.

Reply:
Should this be more generic, even, so instead of `Dialer()` we have something else like `ExtraDialOptions() []grpc.DialOption`?

If not, I'm fine with this.  I  highly doubt anyone is implementing a `Dialer` on a custom xds credentials.

Reply:
> Should this be more generic, even, so instead of Dialer() we have something else like ExtraDialOptions() []grpc.DialOption?

I think this depends on how we think the API and registered credentials might change/evolve in the future, if any. So far, this is the first use of this. From my perspective, if there is a need to expand and generalize in the future, then we can consider it at that time according to YAGNI principle. What are your thoughts @easwars?

Reply:
I think we can handle that as part of https://github.com/grpc/grpc-go/issues/7661. Thanks.